src/dioptra/restapi/experiment/controller.py

```diff
@@ -55,13 +55,35 @@ def __init__(self, *args, experiment_service: ExperimentService, **kwargs) -> No
 
     @login_required
     @responds(schema=ExperimentSchema(many=True), api=api)
-    def get(self) -> List[Experiment]:
-        """Gets a list of all registered experiments."""
+    def get(self) -> Response:
+        """Gets a page or list of all registered experiments."""
         log: BoundLogger = LOGGER.new(
             request_id=str(uuid.uuid4()), resource="experiment", request_type="GET"
         )  # noqa: F841
         log.info("Request received")
-        return self._experiment_service.get_all(log=log)
+
+        index = request.args.get("index", -1, type=int)
+        page_length = request.args.get("page_length", -1, type=int)
+
+        if -1 in (index, page_length):
+            return self._experiment_service.get_all(log=log)  # type: ignore
+
+        data = self._experiment_service.get_page(
+            index=index, page_length=page_length, log=log
+        )
+
+        is_complete = True if Experiment.query.count() <= index + page_length else False
+
+        return jsonify(
```

Comment:
This should probably be a schema defined in `experiment/schema.py`. It's possible that we can have a page schema which uses the experiment schema via `fields.Nested()`. There might be another design pattern we should use, so I'll leave it up to @jkglasbrenner to decide.